flow/embedded_views.h

```diff
@@ -253,6 +253,17 @@ class ExternalViewEmbedder {
   // This is called after submitting the embedder frame and the surface frame.
   virtual void FinishFrame();
 
+  // This should only be called after |SubmitFrame|.
+  // This method provides the embedder a way to do additional tasks after
+  // |SubmitFrame|. After invoking this method, the current task on the
+  // TaskRunner should end immediately.
+  //
+  // For example on the iOS embedder, threads are merged in this call.
+  // A new frame on the platform thread starts immediately. If the GPU thread
+  // still has some task running, there could be two frames being rendered
+  // concurrently, which causes undefined behaviors.
+  virtual void EndFrame(fml::RefPtr<fml::GpuThreadMerger> gpu_thread_merger) {}
```

Comment:
q:  The description of this method seems similar similar to `FinishFrame`. Is it possible to have a single method for both?

Reply:
we currently only call EndFrame during thread merging process. It doesn't seem to be doing the same thing with we are doing in `FinishFrame`.

Reply:
This is used particularly in the thread merging scenario. `FinishFrame` is actually not the very end of the frame. But the `EndFrame` marks the very end of the frame.